### PR TITLE
Fix Tasty errors

### PR DESCRIPTION
**Pull request by odersky**
Fixes one of the errors reported in #1202. If I can fix more errors, I'll try to push those fixes on this PR later. Review by @VladimirNik .

### CODE REVIEW

**Comment by odersky** [comment]
Now the other errors reported in #1202 should also be fixed.

**Comment by VladimirNik** [comment]
@odersky Thank you! LGTM

File: src/dotty/tools/dotc/FromTasty.scala [hunk 1]
```diff
@@ -64,6 +64,9 @@ object FromTasty extends Driver {
   }
 
   class ReadTastyTreesFromClasses extends FrontEnd {
+
+    override def isTyper = false
```
Comment (DarkDimius) [inline src/dotty/tools/dotc/FromTasty.scala]: Unfortunately, this is not enough. 
A lot of code checks for `Phases.isAfterTyper` that is entirely independent from this method

File: src/dotty/tools/dotc/transform/Pickler.scala [hunk 2]
```diff
@@ -68,7 +68,7 @@ class Pickler extends Phase {
   private def pickleSourcefile(pickler: TastyPickler, source: SourceFile): Unit = {
     val buf = new TastyBuffer(10)
     pickler.newSection("Sourcefile", buf)
-    buf.writeNat(pickler.nameBuffer.nameIndex(source.file.path).index)
+    buf.writeNat(pickler.nameBuffer.nameIndex(source.file.canonicalPath).index)
```
Comment (DarkDimius) [inline src/dotty/tools/dotc/transform/Pickler.scala]: This may leak private knowledge of folder structure.
Why is relative path not enough?

Comment (odersky) [inline src/dotty/tools/dotc/transform/Pickler.scala]: ExpandPrivate contains an assertion that compares two source files for equality. It only works when we canonicalize the paths. We could weaken the assertion and demand e.g. that the paths have a common non-trivial suffix. I'll make that change.

File: src/dotty/tools/dotc/FromTasty.scala (hunk 1 above)
Comment (odersky) [inline src/dotty/tools/dotc/FromTasty.scala]: True. But it does not matter for anyway. The change did not affect the fixes. I'll add a comment to explain better what isTyper is.

**Comment by DarkDimius** [comment]
I'm not sure if change proposed in 5aa59ac17e6f48719f8b52048e0b86570e9399e3 is good for tooling.

It makes it very complicated for an external tool to match TASTY files with source files. As the tool would be required to be able to read the TASTY section.

Though it also has bright sides, as it would simplify storing TASTY for classes originating from different source files.

**Comment by odersky** [comment]
If necessary we could provide a method that returns the top level classes of a tasty file. Then we can read the source files from them. I think it is an advantage that A tasty file is not tied to one source.

**Comment by VladimirNik** [comment]
I added commit with updates for TASTY tests (for dotty/src)

**Pull request merged**